datafusion/src/physical_plan/coercion_rule/binary_rule.rs

```diff
@@ -77,7 +77,7 @@ pub(crate) fn coerce_types(
 }
 
 fn comparison_eq_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
-    if lhs_type == rhs_type {
+    if lhs_type == rhs_type && !is_dictionary(lhs_type) {
```

Comment:
Can you add `TODO` or some comments for this change?

Reply:
I will do so -- it is a good point

Reply:
in e27110fd1